packages/myst-toc/src/toc.ts

```diff
@@ -1,21 +1,193 @@
-import type { TOC } from './types.js';
-import schema from './schema.json';
-import _Ajv from 'ajv';
+import type { TOC, Entry, FileEntry, URLEntry, PatternEntry, ParentEntry } from './types.js';
+
+import type { ValidationOptions } from 'simple-validators';
+import {
+  defined,
+  incrementOptions,
+  validateBoolean,
+  validateEnum,
+  validateList,
+  validateNumber,
+  validateObjectKeys,
+  validateString,
+  validationError,
+} from 'simple-validators';
+
+const COMMON_ENTRY_KEYS = ['title', 'hidden', 'numbering', 'id', 'part', 'class'];
+
+function validateCommonEntry(entry: any, opts: ValidationOptions): Entry | undefined {
```

Comment:
```suggestion
function validateCommonEntry(entry: Record<string, any>, opts: ValidationOptions): Entry {
```
Just for slight type safety/accuracy

Reply:
Interesting - I left this as `| undefined` to give us future scope to change that ... but, it's actually not public so there's no need!